src/download.rs

```diff
@@ -291,6 +285,63 @@ pub fn unpack_bins(dir: &Path, dst_dir: &Path) -> Result<Vec<PathBuf>> {
     Ok(downloaded)
 }
 
+/// write Ok(Response) to provided writer with progress bar displaying writing status
+pub fn write_response_with_progress_bar<W: Write>(
+    response: Response,
+    writer: &mut W,
+) -> Result<()> {
+    let total_size = response
+        .header("Content-Length")
+        .and_then(|s| s.parse::<u64>().ok())
+        .unwrap_or(0);
+    let mut downloaded_size = 0;
+    let mut buffer = [0; 8192];
+    let progress_bar = ProgressBar::new(total_size);
+    progress_bar.set_style(
+                ProgressStyle::default_bar()
+                    .template("[{elapsed_precise}] [{bar:40.cyan/blue}] {bytes}/{total_bytes} {bytes_per_sec} ({eta}) - {msg:.cyan}")
```

Comment:
I don't think we need to show the bytes per sec, and it's nicer to have "Download complete" in green

```suggestion
                    .template("[{elapsed_precise}] [{bar:40.cyan/blue}] {bytes}/{total_bytes} ({eta}) - {msg:.green}")
```